web/app/js/components/ResourceList.jsx

```diff
@@ -30,29 +30,39 @@ export default class ResourceList extends React.Component {
   }
 
   componentDidMount() {
-    this.loadFromServer();
-    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+    this.startServerPolling();
   }
 
-  componentWillReceiveProps() {
+  componentWillReceiveProps(newProps) {
     // React won't unmount this component when switching resource pages so we need to clear state
-    this.api.cancelCurrentRequests();
-    this.setState(this.getInitialState());
+    this.stopServerPolling();
+    this.setState(this.getInitialState(newProps));
+    this.startServerPolling(newProps.resource);
   }
 
   componentWillUnmount() {
+    this.stopServerPolling();
+  }
+
+  startServerPolling(resource) {
+    this.loadFromServer(resource);
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
+
+  stopServerPolling() {
     window.clearInterval(this.timerId);
     this.api.cancelCurrentRequests();
   }
 
-  loadFromServer() {
+  loadFromServer(resource) {
+    resource = resource || this.props.resource;
```

Comment:
naive question: is it possible to do:
```js
componentDidMount() {
  this.startServerPolling(this.props.resource);
}
```

...and then avoid this check for null `resource`?

Reply:
yep, can do!